src/toolbar/Toolbar.ts

```diff
@@ -0,0 +1,158 @@
+import { debounce } from '@dojo/core/util';
+import { Dimensions } from '@dojo/widget-core/meta/Dimensions';
+import { DNode, WidgetProperties } from '@dojo/widget-core/interfaces';
+import { ThemedMixin, theme } from '@dojo/widget-core/mixins/Themed';
+import { v, w } from '@dojo/widget-core/d';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+
+import SlidePane, { Align } from '../slidepane/SlidePane';
+import * as css from './styles/toolbar.m.css';
+import * as iconCss from '../common/styles/icons.m.css';
+
+/**
+ * @type ToolbarProperties
+ *
+ * Properties that can be set on a Toolbar component
+ *
+ * @property actions           Action elements to show in the toolbar
+ * @property collapseWidth     Width at which to collapse actions into a SlidePane
```

Comment:
I feel strongly that we shouldn't _always_ be inverting control to the parent for every piece of UI functionality. It's both tedious and common enough to hide action items in a SlidePane that our opinionated component should provide this, while also allowing easy extension. In practice, using a property like `collapseWidth` instead of `collapsed` and internalizing window resizing logic felt much cleaner.